include/core/adapter/Adapter2D.hpp

```diff
@@ -0,0 +1,69 @@
+#ifndef GENERIC_ADAPTER_H
+#define GENERIC_ADAPTER_H
+
+#include <memory>
+#include <string>
+#include <unordered_map>
+
+#include <boost/geometry.hpp>
+#include <boost/geometry/geometries/geometries.hpp>
+#include <HY_Catchment.hpp>
+#include <HY_HydroLocation.hpp>
+
+using namespace hy_features::hydrolocation;
+class GenericAdapter
+{
+    public:
+
+    typedef long time_step_t;
+
+
+    // create type aliases
+    using Models = std::vector<std::string>;
+    using long = time_point_t;
+    using long = time_delta;
+
+    GenericAdapter(std::string adapter_id, Models ls, Models rs);
+
+    virtual ~Adapter2D();
```

Comment:
The class/constructor and the file/destructor don't have the same name.  I am presuming this is not intended.

Reply:
That would be correct. I will update the branch